mycore-restapi/src/main/java/org/mycore/restapi/v2/access/MCRRestAccessManager.java

```diff
@@ -22,22 +22,45 @@
 import org.mycore.access.MCRAccessManager;
 import org.mycore.access.MCRRuleAccessInterface;
 
+/**
+ * The {@code MCRRestAccessManager} class manages access to REST API resources.
+ */
 public class MCRRestAccessManager {
 
     private static final String REST_API_OBJECT_ID = "restapi:/";
 
+    /**
+     * Checks access to a REST API resource based on a given permission and path.
+     *
+     * This method first checks whether a rule applies to the entire REST API
+     * ({@code REST_API_OBJECT_ID}). If this rule exists and the required permission is granted,
+     * it proceeds to check whether a specific rule exists for the given path.
+     * Based on the rules, it determines whether access to the path is allowed or not.
+     *
+     * @param aclProvider an {@code MCRAccessInterface} used to check permissions
+     * @param permission the permission to be checked
+     * @param path the path of the REST API resource for which access is being checked
+     * @return {@code true} if access is granted, otherwise {@code false}.
+     */
     public static boolean checkRestAPIAccess(MCRAccessInterface aclProvider, String permission, String path) {
-        final MCRAccessInterface acl = MCRAccessManager.getAccessImpl();
-        if (aclProvider.checkPermission(REST_API_OBJECT_ID, permission)) {
-            final String objectId = path.startsWith("/") ? REST_API_OBJECT_ID + path.substring(1)
-                : REST_API_OBJECT_ID + path;
-            if (!(acl instanceof MCRRuleAccessInterface ruleAccess) || ruleAccess.hasRule(objectId, permission)) {
-                return aclProvider.checkPermission(objectId, permission);
-            } else {
-                return true;
-            }
+        final MCRAccessInterface acl = MCRAccessManager.getAccessImpl(); // TODO use aclProvider instead?
```

Comment:
I've had some difficulty recognizing what is describe in the comments in the actual code and did some investigation.

Suggestion for an equivalent implementation (that, I'd argue, is easier to grasp; see Skype chat for details):

```
public static boolean doCheckRestAPIAccess2(MCRAccessInterface acl, MCRAccessInterface aclProvider, String permission, String path) {

    final String objectId = getObjectId(path);

    boolean hasGeneralRule = false;
    boolean hasSpecificRule = false;
    if (acl instanceof MCRRuleAccessInterface ruleAccess) {
        hasGeneralRule = ruleAccess.hasRule(REST_API_OBJECT_ID, permission);
        hasSpecificRule = ruleAccess.hasRule(objectId, permission);
    }

    boolean hasGeneralPermission = aclProvider.checkPermission(REST_API_OBJECT_ID, permission);
    boolean hasSpecificPermission = aclProvider.checkPermission(objectId, permission);

    if (!hasSpecificRule) {
        return hasGeneralPermission;
    }

    if (!hasGeneralRule) {
        return hasSpecificPermission;
    }

    return hasGeneralPermission && hasSpecificPermission;

}
```

Reply:
Thanks for your review. I used the original code as a basis - I just put a rule in front of it, if you like. Actually, your code seems to be equivalent and is easier to understand. Thank you.